lib/find-python.js

```diff
@@ -18,16 +18,15 @@ PythonFinder.prototype = {
   log: logWithPrefix(log, 'find Python'),
   argsExecutable: ['-c', 'import sys; print(sys.executable);'],
   argsVersion: ['-c', 'import sys; print("%s.%s.%s" % sys.version_info[:3]);'],
-  semverRange: '2.7.x || >=3.5.0',
+  semverRange: '>=3.5.0',
 
   // These can be overridden for testing:
   execFile: cp.execFile,
   env: process.env,
   win: win,
   pyLauncher: 'py.exe',
   winDefaultLocations: [
-    path.join(process.env.SystemDrive || 'C:', 'Python37', 'python.exe'),
-    path.join(process.env.SystemDrive || 'C:', 'Python27', 'python.exe')
+    path.join(process.env.SystemDrive || 'C:', 'Python37', 'python.exe')
```

Comment:
36, 38, 39 are also valid.

Reply:
I think those default locations are not valid for Python 3.5 and higher anymore. See e.g. https://stackoverflow.com/questions/44749636/what-path-to-install-python-3-6-to-on-windows

For system-wide installs they are now in `%ProgramFiles%` and per user in `%LocalAppData%\Programs`.
Maybe you can look in both those directories and loop with decreasing the python version number, so that you find the highest installed version?
Though it gets complicated, so maybe drop it completely and just tell the user to set the path to python correctly?

Reply:
> For system-wide installs they are now in %ProgramFiles% and per user in %LocalAppData%\Programs. 👍

Reply:
@mfonville is right. `'C:\Python37'\python.exe` isn't the real format for the default install path for Python 3.

I can confirm from testing the installers, that the format would be one of the following:

- `C:\Users\[user]\AppData\Local\Programs\Python\Python39`
- `C:\Program Files\Python39`
- `C:\Users\[user]\AppData\Local\Programs\Python\Python39-32`
- `C:\Program Files (x86)\Python39-32`

Here is an example branch using those paths for all supported Python versions (3.6-3.9): https://github.com/DeeDeeG/node-gyp/compare/find-python-only-find-python-3...DeeDeeG:ae4e0b4c7a1e9a5c9a0b877c2ed737fe5c6332cc

This is not much slower at all, so I think it would be worth adding.

(Unfortunately I can't figure out how to adjust the tests to pass with this change. If anyone can assist with that, it would be much appreciated.)

Reply:
I have fixed the tests when using the real Python 3 default install paths. ([Diff](https://github.com/DeeDeeG/node-gyp/compare/find-python-only-find-python-3...update-Python-fallback-paths-for-Windows), [CI](https://github.com/DeeDeeG/node-gyp/actions/runs/576058726)).

@cclauss should I add that to this PR? (Or perhaps wait for a follow-up PR?) Thanks.

Also: Python 3.10 will be out [in October](https://www.python.org/dev/peps/pep-0619/#release-schedule), and the [alpha release](https://www.python.org/downloads/release/python-3100a5/) seems to work with `node-gyp` in my testing. I could add 3.10 install paths as well.

Reply:
Please put those changes for the supported Pythons (3.6-3.9) in this PR but let's hold off on 3.10 until it is released.  We do not want to support any Python 3.10 breakage until it is officially released.  Thanks massively for your awesome work!

Reply:
The system-wide Python install path for 32-bit Windows is `C:\Program Files\Python39-32`. (There is no `Program Files (x86)` on 32-bit Windows, only `Program Files`.)

 That's one more path to check per minor version.